packages/regenesis-surgery/scripts/surgery.ts

```diff
@@ -0,0 +1,346 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import { ethers } from 'ethers'
+import * as dotenv from 'dotenv'
+import * as UNISWAP_V3_CORE_BUILD_INFO from '../contracts/v3-core/artifacts/build-info/3ae1a3bfa113927fd93fec3ee383c82b.json'
+import { abi as UNISWAP_V3_FACTORY_ABI } from '../contracts/v3-core/artifacts/contracts/UniswapV3Factory.sol/UniswapV3Factory.json'
+import {
+  abi as UNISWAP_V3_POOL_ABI,
+  deployedBytecode as UNISWAP_V3_POOL_DEPLOYED_BYTECODE,
+} from '../contracts/v3-core/artifacts/contracts/UniswapV3Pool.sol/UniswapV3Pool.json'
+
+interface ChainState {
+  [address: string]: {
+    balance: string
+    nonce: number
+    root: string
+    codeHash: string
+    code?: string
+    storage?: {
+      [key: string]: string
+    }
+  }
+}
+
+interface StateDump {
+  root: string
+  accounts: ChainState
+}
+
+// TODO: Get these two values from input?
+const UNISWAP_V3_FACTORY_ADDRESS = '0x1f98431c8ad98523631ae4a59f267346ea31f984'
+const UNISWAP_NFPM_ADDRESS = '0xc36442b4a4522e871399cd717abdd847ab11fe88'
+// TODO: Get these two values from something like ethereumjs
+const DEFAULT_STORAGE_ROOT =
+  '56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421'
+const DEFAULT_CODE_HASH =
+  'c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470'
+const EOA_CODE_HASHES = [
+  'a73df79c90ba2496f3440188807022bed5c7e2e826b596d22bcb4e127378835a',
+  'ef2ab076db773ffc554c9f287134123439a5228e92f5b3194a28fec0a0afafe3',
+]
+// TODO: Ideally get these values from the contract itself
+const OLD_UNISWAP_POOL_INITCODE_HASH =
+  '0x0c231002d0970d2126e7e00ce88c3b0e5ec8e48dac71478d56245c34ea2f9447'
+const NEW_UNISWAP_POOL_INITCODE_HASH =
+  '0xe34f199b19b2b4f47f68442619d555527d244f78a3297ea89325f843f87b8b54'
+
+// TODO: Maybe Uniswap already has a function for this?
+// Otherwise just add comments here.
+const getUniPoolAddress = (
+  token0: string,
+  token1: string,
+  fee: number | ethers.BigNumber,
+  factoryAddress: string,
+  initCodeHash: string
+) => {
+  if (ethers.BigNumber.from(token0).gt(ethers.BigNumber.from(token1))) {
+    ;[token0, token1] = [token1, token0]
+  }
+  return ethers.utils
+    .getCreate2Address(
+      factoryAddress,
+      ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(
+          ['address', 'address', 'uint24'],
+          [token0, token1, fee]
+        )
+      ),
+      initCodeHash
+    )
+    .toLowerCase()
+}
+
+const toHex32 = (val: string | number | ethers.BigNumber) => {
```

Comment:
Could replace this with `const toHex32 = (val) => ethers.utils.hexZeroPad(val, 32)` to avoid reimplementing the padding logic